exporter_test.go

```diff
@@ -153,3 +153,21 @@ func (ml *mockStatsDTCPListener) handlePacket(packet []byte, e chan<- Events) {
 	}
 	ml.handleConn(sc, e)
 }
+
+func TestEscapeMetricName(t *testing.T) {
+	scenarios := map[string]string{
+		"clean":                   "clean",
+		"0starts_with_digit":      "_0starts_with_digit",
+		"with_underscore":         "with_underscore",
+		"with.dot":                "with_dot",
+		"with😱emoji":              "with_emoji",
+		"with.*.multiple":         "with___multiple",
+		"test.web-server.foo.bar": "test_web_server_foo_bar",
+	}
+
+	for in, want := range scenarios {
+		if got := escapeMetricName(in); want != got {
```

Comment:
You could also just reuse Prometheus' strutil.SanitzeLabelName.

Reply:
good point, but that doesn't do the prefixing if the first character is a digit. should it?

I don't want to change behavior at this point, so I'll stick with the function as it is right now.

Reply:
Actually, it's not a guide idea to use it for this use case. It's not made to sanitize metric names, and not worth adding a dependency. Nevermind.